src/components/Calculator.js

```diff
@@ -1,172 +1,188 @@
+      },
+    );
+  };
+
+  const handleResults = (e) => {
+    const calculation = calculate(state, e.target.innerHTML);
+
+    setState(calculation);
+  };
+
+  const { total, next, operation } = state;
 
+  return (
+    <div className="calculator">
+
+      <div className="results general">
+        <span className="results-span">
+          { total || 0}
+          { operation }
+          { next }
+        </span>
       </div>
-    );
-  }
-}
+
+      <Button
+        content="AC"
+        classname="ac general grayscale button"
+        onClick={handleResults}
+      />
+
+      <Button
+        content="+/-"
+        classname="plusminus general grayscale button"
+        onClick={handleResults}
+      />
+
+      <Button
+        content="%"
+        classname="modulus general grayscale button"
+        onClick={operationHandler}
+      />
+
+      <Button
+        content="÷"
+        classname="division general orange button"
+        onClick={operationHandler}
+      />
+
+      <Button
+        content="7"
+        classname="seven general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="8"
+        classname="eight general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="9"
+        classname="nine general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="x"
+        classname="multiplication general orange button"
+        onClick={operationHandler}
+      />
+
+      <Button
+        content="4"
+        classname="four general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="5"
+        classname="five general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="6"
+        classname="six general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="-"
+        classname="minus general orange button"
+        onClick={operationHandler}
+      />
+
+      <Button
+        content="1"
+        classname="one general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="2"
+        classname="two general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="3"
+        classname="three general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="+"
+        classname="plus general orange button"
+        onClick={operationHandler}
+      />
+
+      <Button
+        content="0"
+        classname="zero general grayscale button"
+        onClick={numbersHandler}
+      />
+
+      <Button
+        content="."
+        classname="dot general grayscale button"
+        onClick={handleResults}
+      />
+
+      <Button
+        content="="
+        classname="equal general orange button"
+        onClick={handleResults}
+      />
```

Comment:
- [ ] [OPTIONAL] Instead of creating each button manually you could create an array with objects, that hold that button data, and then display the buttons with `map` something like this:
```javascript
    {buttonsArr.map((button) => (
       <Button content={button. content} className={button.className} onClick={button.event} 
    ))

```